pkg/fragment/definition.go

```diff
@@ -43,19 +62,44 @@ func WithTimingLabel(timingLabel string) DefinitionOption {
 	}
 }
 
-func (d *Definition) UrlWithParams(parameters url.Values) string {
+func (d *Definition) DynamicParts() []string {
+	return d.dynamicParts
+}
+
+func (d *Definition) UrlWithParams(parameters url.Values) *url.URL {
 	// This is already parsed before constructing the url in server.go, so we ignore errors
 	targetUrl, _ := url.Parse(d.Url)
 	targetUrl.RawQuery = parameters.Encode()
 
-	return targetUrl.String()
+	return targetUrl
 }
 
-func (d *Definition) IntoRequestable(params url.Values) multiplexer.Requestable {
+func (d *Definition) Requestable(target *url.URL, pathParams map[string]string, query url.Values) (*Request, error) {
+	request := *target // clone the url
+
+	var path strings.Builder
+
+	for _, part := range d.routeParts {
+		path.WriteByte('/')
+
+		if strings.HasPrefix(part, ":") {
+			if replacement, ok := pathParams[part]; ok {
+				path.WriteString(replacement)
+			} else {
+				return nil, fmt.Errorf("no url replacement found for %s in %s", part, d.Path)
```

Comment:
What do you think about testing this from `pkg/fragment/definition_test.go` ? Or was it covered in tests in other packages?

Reply:
I like it, added in https://github.com/BlakeWilliams/viewproxy/pull/71/commits/d888a6c673e546f8fb4522ba4a983adbf44a6ae7